lecture-notes/2-vcs.md

```diff
@@ -1989,8 +1989,141 @@ $
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 14)     return left / right;
+2d660302 (Thomas Hebb 2024-02-21 23:07:15 -0500 15)   case  '%':
+2d660302 (Thomas Hebb 2024-02-21 23:07:15 -0500 16)     return left % right;
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 17)   }
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 18)   fprintf(stderr, "Unrecognized op `%c'.\n", op);
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 19)   exit(EXIT_FAILURE);
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 20) }
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 21)
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 22) int main(int argc, char **argv) {
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 23)   if (argc != 4) {
+2d660302 (Thomas Hebb 2024-02-21 23:07:15 -0500 24)     fprintf(
+2d660302 (Thomas Hebb 2024-02-21 23:07:15 -0500 25)         stderr,
+2d660302 (Thomas Hebb 2024-02-21 23:07:15 -0500 26)         "Usage: %s <num> <op> <num>\nWhere <op> is one of +, -, *, /, %%.\n",
+2d660302 (Thomas Hebb 2024-02-21 23:07:15 -0500 27)         argv[0]);
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 28)     return EXIT_FAILURE;
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 29)   }
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 30)   const char *left_str = argv[1];
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 31)   const char *op_str = argv[2];
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 32)   const char *right_str = argv[3];
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 33)   errno = 0;
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 34)   int left = strtol(left_str, NULL, 10);
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 35)   if (errno != 0) {
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 36)     perror(argv[0]);
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 37)     return EXIT_FAILURE;
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 38)   }
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 39)   if (op_str[1] != '\0') {
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 40)     fprintf(stderr, "Op must be one character. Got `%s'.\n", op_str);
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 41)     return EXIT_FAILURE;
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 42)   }
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 43)   errno = 0;
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 44)   int right = strtol(right_str, NULL, 10);
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 45)   if (errno != 0) {
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 46)     perror(argv[0]);
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 47)     return EXIT_FAILURE;
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 48)   }
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 49)   int result = calc(left, op_str[0], right);
+1838eeae (Thomas Hebb 2024-01-31 21:07:08 -0500 50)   fprintf(stdout, "%d\n", result);
+^3ba2a93 (Thomas Hebb 2024-01-30 22:57:04 -0500 51)   return 0;
+^3ba2a93 (Thomas Hebb 2024-01-30 22:57:04 -0500 52) }
+$ 
+```
```

Comment:
might as well include a cultural message as well about blameless blah blah

Reply:
Not sure how to work that in, can you suggest something?

Reply:
"Please don't take the command name 'blame' as an excuse to be mean to your coworkers. Be civil." Maybe link to blameless postmortem